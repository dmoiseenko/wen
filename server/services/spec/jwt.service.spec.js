const jwtService = require('../jwt.service.js');


it('should generate token and verify it', async () => {
  const data = 'data';

  const actual = await jwtService.generateToken(data);

  expect(actual).toMatchSnapshot();
});

it('should generate token and verify it', async () => {
  const token = 'eyJhbGciOiJIUzI1NiJ9.OWZiNTIyNmRjMTVjZWU1ZmMxODhiYmVmODBkZjc3NDU.dqH5DqCy5XeciTE9D7yIdLeEVel8HN5dcwH_waQP7as';

  const actual = await jwtService.verifyTokenAndReturnUserData(token);

  expect(actual).toMatchSnapshot();
});

it('should throw InvalidTokenError if token is invalid', async () => {
  try {
    await jwtService.verifyTokenAndReturnUserData('token');
  } catch (err) {
    expect(err.name).toEqual('InvalidTokenError');
  }
});
