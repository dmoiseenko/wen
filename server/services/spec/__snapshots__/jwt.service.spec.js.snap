// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate token and verify it 1`] = `"eyJhbGciOiJIUzI1NiJ9.MDQzMmQ5ODU1Zjdk.KiAacE8kqxVSecWzuwDeLLp9EB0z19VrZiIuI7_iiQ0"`;

exports[`should generate token and verify it 2`] = `"data"`;
